### ISSUE TITLE
RESTORE ... WITH into_schema 

### ISSUE THREAD

**Issue by data-matt** [cockroachdb/cockroach#78941]
**Is your feature request related to a problem? Please describe.**

In light of the change that the following is deprecated:
`ALTER TABLE test RENAME TO new_table_name`

If someone was to restore a table using `into_db` to restore a table to a different target database, they wouldn't be able to move that table from the target database, to the original database. 

**Describe the solution you'd like**
Please provide an option to restore the table into the same database with a different schema or name, so that you can restore another copy of a table to the original database.

e.g
```
RESTORE bank.customers FROM '2021/03/23-213101.37' IN 's3://{bucket_name}/{path/to/backup}?AWS_ACCESS_KEY_ID={key_id}&AWS_SECRET_ACCESS_KEY={access_key}'
WITH into_schema = 'newschema';
```

**Additional context**
[Add any other context or screenshots about the feature request here.](https://cockroachlabs.slack.com/archives/C0168LW5THS/p1648458816145989?thread_ts=1648228225.815339&cid=C0168LW5THS)

**Comment by knz** [comment]
I think there's a misunderstanding: ALTER RENAME TO is still supported and will continue to be (like in postgres).

The only thing that's limited is the ability to rename *across databases*.

- rename table within schema: ALTER  RENAME TO works
- rename table across user-defined schemas: ALTER SET SCHEMA  (not RENAME TO)
- rename table across databases: we need to talk about this. It's midlly possible we will still able to support  a way to do this without backup/restore, just in a different way than pg

**Comment by data-matt** [comment]
@knz, that is understood. The concern comes from the point of view that a customer currently can restore a table to a different target database for staging purposes and move it. If we remove the ability to rename a table to move it from one database to another, then this type of restore isn't viable for that scenario any more. So, if we provide a way to restore a table to a different schema, we can stage tables in under a different schema right?

**Comment by knz** [comment]
yes. But I also wonder if we can support moving a table from one db to another using another mechanism than a simple rename? maybe @ajwerner has opinions on this.

**Comment by data-matt** [comment]
Don't mind 👍  We just need to be cognisant that we still leave customers with a method to restore a table some where else other.. or with a different name. Otherwise some customers will feel the pain of regression.
